### PR TITLE
Regression: compile-app-settings with --debug flag should ignore eslint errors

### PR DESCRIPTION
**Pull request by kennsippell**
This is a fix for https://github.com/medic/cht-conf/issues/533

When I was adding a regression test, I noticed a few problems with the tests in compile-app-settings.spec.js:

1. The "failure" scenarios expect any exception to throw. That said, actual exception thrown is the error `No configuration defined` because the content being tested (`testDir`) is not properly used.
2. When the correct folder is loaded, the tests now throw the exception `state was not initialized` via environment.js. This is a failure in the test setup.
2. Tests like `sms-modules/invalid-files` are failing because of a missing `.eslintrc` -- not for the intended scenario being tested
4. It's impossible for the eslint scenarios to ever fail because our `.eslintignore` file includes `test/data/**` - which causes all eslint validation to be skipped

So the propose changes in this PR are focused on fixing https://github.com/medic/cht-conf/issues/533 + adding a regression test + and fixing the existing tests. Specifically:

1. Fix for initial bug https://github.com/medic/cht-conf/issues/533. One-liner in `package-lib.js`. When the eslint npm package was upgraded via https://github.com/medic/cht-conf/issues/414, the crash stack changed and is no longer recognized.
2. Add regression test for https://github.com/medic/cht-conf/issues/533
2. Update compile-app-settings.spec rejection cases to expect a specific error string consistent with the scenario they intend to test. Then, fix tests which are not passing for the correct reasons.
3. Allow compile-app-settings.spec to test eslint scenarios by disabling .eslintignore
5. I also refactored the tests in compile-app-settings.spec to loop through a list of "scenarios". Seemed a little cleaner, but I'm open to reverting that refactor.

### CODE REVIEW

File: test/fn/compile-app-settings.spec.js [hunk 1]
```diff
@@ -4,111 +4,152 @@ const sinon = require('sinon');
...
+        } catch (err) {
+          expect(err.toString()).to.include(scenario.error);
+        }
+      } else {
+        await promiseToExecute;
+        const actual = JSON.parse(JSON.stringify(writeJson.args[0][1]));
+        const expected = JSON.parse(fs.read(`${pathToTestProject}/../app_settings.expected.json`));
+        actual.tasks.rules = expected.tasks.rules = '';
+        actual.contact_summary = expected.contact_summary = '';
+        expect(actual).to.deep.eq(expected);
```
Comment (dianabarsan) [inline test/fn/compile-app-settings.spec.js:151]:
```suggestion
        const actual = writeJson.args[0][1];
        const expected = JSON.parse(fs.read(`${pathToTestProject}/../app_settings.expected.json`));
        expect(actual).excludingEvery(['rules', 'contact_summary', 'schedules']).to.deep.eq(expected);
```

You'd need to also plug in `chai-exclude` somewhere up top (or ideally globally):
```
require('chai').use(require('chai-exclude'));
```

File: test/fn/compile-app-settings.spec.js [hunk 2]
```diff
@@ -4,111 +4,152 @@ const sinon = require('sinon');
...
+      sinon.stub(environment, 'extraArgs').get(() => scenario.extraArgs);
+
+      const promiseToExecute = compileAppSettings.execute({ skipEslintIgnore: true });
+      if (scenario.error) {
+        try {
+          await promiseToExecute;
+          assert.fail('Expected execute() to throw');
+        } catch (err) {
+          expect(err.toString()).to.include(scenario.error);
+        }
```
Comment (dianabarsan) [inline test/fn/compile-app-settings.spec.js:144]: I find `chai-as-promised` to provide a really nice way to test promises that reject. It's not a dependency right now, but it's easy to add. With `chai-as-promised` these lines can be written as: 
```suggestion
        await expect(promiseToExecute).to.be.rejectedWith(Error, scenario.error);
```

**Review by dianabarsan — COMMENTED** [review]
Thanks a lot for taking care of the broken tests and cleaning up the scenarios section, which is much easier to follow now. 

I'm approving since this can work as is, but did leave some suggestions in about compressing the test code even more.

**Review by dianabarsan — APPROVED** [review]

File: test/fn/compile-app-settings.spec.js (hunk 2 above)
Comment (kennsippell) [reply test/fn/compile-app-settings.spec.js:144]: nice nice

**Comment by garethbowen** [comment]
@kennsippell Looks like this passed AT... could you please fix the conflict and then merge? (remember the commit format!).

**Pull request merged**